testing/simapp/sim_bench_test.go

```diff
@@ -17,30 +21,39 @@ import (
 // /usr/local/go/bin/go test -benchmem -run=^$ github.com/cosmos/ibc-go/v7/testing/simapp -bench ^BenchmarkFullAppSimulation$ -Commit=true -cpuprofile cpu.out
 func BenchmarkFullAppSimulation(b *testing.B) {
 	b.ReportAllocs()
-	config, db, dir, logger, _, err := SetupSimulation("goleveldb-app-sim", "Simulation")
+
+	config := simcli.NewConfigFromFlags()
+	config.ChainID = SimAppChainID
+
+	db, dir, logger, skip, err := simtestutil.SetupSimulation(config, "goleveldb-app-sim", "Simulation", simcli.FlagVerboseValue, simcli.FlagEnabledValue)
 	if err != nil {
 		b.Fatalf("simulation setup failed: %s", err.Error())
 	}
 
+	if skip {
+		b.Skip("skipping benchmark application simulation")
+	}
+
 	defer func() {
-		db.Close()
-		err = os.RemoveAll(dir)
-		if err != nil {
-			b.Fatal(err)
-		}
+		require.NoError(b, db.Close())
+		require.NoError(b, os.RemoveAll(dir))
 	}()
 
-	app := NewSimApp(logger, db, nil, true, map[int64]bool{}, DefaultNodeHome, FlagPeriodValue, MakeTestEncodingConfig(), simtestutil.EmptyAppOptions{}, interBlockCacheOpt(), baseapp.SetChainID(SimAppChainID))
+	appOptions := make(simtestutil.AppOptionsMap, 0)
+	appOptions[flags.FlagHome] = DefaultNodeHome
+	appOptions[server.FlagInvCheckPeriod] = simcli.FlagPeriodValue
+
+	app := NewSimApp(logger, db, nil, true, appOptions, interBlockCacheOpt())
```

Comment:
When I run `make test-sim-benchmark` (and also `make test-sim-profile`) I get the error `panic: invalid chain-id on InitChain; expected: , got: simulation-app`. I it's because the `chainID` is not set in base app. think we need to pass to `NewSimApp` the extra option `baseapp.SetChainID(SimAppChainID)`, as we did before.

Reply:
will investigate.  I think we may want to set it elsewhere, like in the test...

Reply:
turns out the trick was to not set it at all...

Reply:
benchmark is good, trying profile

Reply:
profile is good now too, nice catch!